src/dmd/errors.d

```diff
@@ -20,6 +20,7 @@ import dmd.globals;
 import dmd.root.outbuffer;
 import dmd.root.rmem;
 import dmd.console;
+import std.conv;
```

Comment:
the compiler is not allowed to depend on the standard library.

Reply:
Ok, I didn't know that!

Reply:
Ok, I didn't know that.
Can you help me by describing how can I convert char* to string without using standard library?

Reply:
You can use the C standard library. Bindings to that is available in `core.stdc`.

Reply:
```
char* p =  ...;
string s = p[0 .. strlen(s)];
```
but the use of `strlen` is discouraged.

But it seems you could make the error header parameter to `verrorPrint` a string as `verrorPrint` is private. It also seems unnecessary for the header to take up the same amount of space as a string since it seems that at a glance it only ever gets called with `Error: `, `       ` (for supplemental), `Warning: `, and `Deprecation: ` an enum. 

`tmp.peekChars` can be related with `tmp.peekString`